File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numset = new HashSet<Integer>();
        for(int num: nums){
            numset.add(num);
        }
        int ms = 0;
        for(int num : numset){
            if(!numset.contains(num-1)){
                int cs = 1;
                while(numset.contains(num+1)){
                    cs += 1;
                    num += 1;
                }
                ms = Math.max(cs, ms);
            }
        }
        return ms;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = set(nums)
        ms = 0
        for num in nums:
            if num+1 not in nums:
                cs = 1
                while num-1 in nums:
                    cs += 1
                    num -= 1
                ms = max(ms, cs)
        return ms
    